ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -458,10 +462,16 @@ def install_replica(master, replica, setup_ca=True, setup_dns=False,
     if master_authoritative_for_client_domain(master, replica):
         args.extend(['--ip-address', replica.ip])
 
-    args.extend(extra_args)
+    ntp_args = [arg for arg in replica_args if "-ntp" in arg]
```

Comment:
I do not like this added logic in a simple utility method. It is very unobvious, that client and replica options should be joined in one list and that this function will mangle this list to make it "work good". In our tests sometimes we do not want it to work, we want it to fail and in this case options must be passed exactly as specified. 
So if you absolutely need to pass ntp options to client installation, I suggest adding another parameter like extra_client_args. 
But as this new feature is used only in one test (test_server_client_install_without_options)  which has outstanding issue that should be resolved, maybe this change will not be needed.

Reply:
With changes for chrony we introduced this behavior:

We can not update chrony configuration while promoting client to replica therefore we have to something with it.

As our test task to install replica is doing the same this task should handle such situation which users of FreeIPA would have to.

I agree that it might be a good idea to have separate extra argument for client but having task to "mangle" argument and have it properly commented inline and documented we may spare some struggles while forgetting that replica promotion will not proceed with NTP options.

I am not only trying to fix test failure but to reflect behavioral changes to our tasks.